pkg/alertmanager/multitenant.go

```diff
@@ -410,6 +434,19 @@ func createMultitenantAlertmanager(cfg *MultitenantAlertmanagerConfig, fallbackC
 	return am, nil
 }
 
+type handlerForGRPCServer struct {
```

Comment:
This mechanism and entire "AlertmanagerServer -> am.grpcServer -> handlerForGRPCServer -> internal serveHTTPHelper" flow should be documented.